File: 1667-find-kth-bit-in-nth-binary-string/find-kth-bit-in-nth-binary-string.cpp
class Solution {
public:
    char findKthBit(int n, int k) {
        string ans;
        ans.push_back('0');
        string flag;
        if(n==1)
        {
            return ans[0];
        }
        for(int i=2;i<=n;i++)
        {
            for(int j=0;j<ans.size();j++)
            {
                if(ans[j]=='0')
                {
                    flag.push_back('1');
                }
                else
                {
                    flag.push_back('0');
                }
            }
            reverse(flag.begin(),flag.end());
            ans.push_back('1');
            ans+=flag;
            flag.clear();
            
        }
        return ans[k-1];
    }
};

File: 1667-find-kth-bit-in-nth-binary-string/find-kth-bit-in-nth-binary-string.java
class Solution {
    public char findKthBit(int n, int k) {
        String s="0";
        StringBuilder s1 = new StringBuilder(s);  
        while(s1.length()<k) 
        {
            StringBuilder s2=new StringBuilder(s1);
            for(int i=0;i<s2.length();i++)
            {
                if(s2.charAt(i)=='0')
                {
                    s2.setCharAt(i,'1');
                }
                else
                {
                    s2.setCharAt(i,'0');
                }
               
                
            } 
            s2.reverse();
            s1.append("1");
            s1.append(s2);
        }
        return s1.charAt(k-1);
    }
}